docs/source/cluster_submission.rst

```diff
@@ -84,4 +84,92 @@ Without any argument the ``--bundle`` option will bundle **all** eligible job-op
+.. _directives:
+
+Submission Directives
+=====================
+
+Executing operations on a cluster environment may involve the specification of resources that are required for said operation
+For this, any :py:class:`~flow.FlowProject` *operation* can be amended with so called *submission directives*.
+For example, to specify that a parallelized operation requires **4** processing units, we would provide the ``np=4`` directive:
+
+.. code-block:: python
+
+    from flow import FlowProject, directives
+    from multiprocessing import Pool
+
+    @FlowProject.operation
+    @directives(np=4)
+    def hello(job):
+        with Pool(4) as pool:
+          print("hello", job)
+
+.. note::
+
+    The directive *np=4* means that the operation **requires** 4 processing units, the operation is not automatically parallelized.
+
+All directives are essentially conventions, the ``np`` directive in particular means that this particular operation requires 4 processors for execution.
+
+.. tip::
+
+    Note that all directives may be specified as callables, e.g. ``@directives(np = lambda job: job.np)``.
+
+Available directives
+--------------------
+
+The following directives are respected by all base templates shipped with **signac-flow**:
+
+.. glossary::
+
+    executable
+      Specify which Python executable should be used to execute this operation.
+      Defaults to the one used to generate the script (:py:attr:`sys.executable`).
+
+    np
+      The total number of processing units required for this operation.
+
+    nranks
+      The number of MPI ranks required for this operation.
+      The command will be prefixed with environment specific MPI command, e.g.: ``mpiexec -n 4``.
+      The value for *np* will default to *nranks* unless specified separately.
+
+    omp_num_threads
+      The number of OpenMP threads.
+      The value for *np* will default to: "nranks x omp_num_threads" unless otherwise specified.
+
+    ngpu
+      The number of GPUs required for this operation.
+
+Execution Modes
+---------------
+
+Using these directives and their combinations allows us to realize the following essential execution modes:
+
+.. glossary::
+
+    serial:
+      ``@flow.directives()``
+
+      This operation is a simple serial process, no directive needed.
+
+    parallelized:
+      ``@flow.directives(np=4)``
+
+      This operation requires 4 processing units.
+
+    MPI parallelized:
+      ``@flow.directives(nranks=4)``
+
+      This operation will be executed on 4 MPI ranks.
+
+    MPI/OpenMP Hybrid:
+      ``@flow.directives(nranks=4, omp_num_threads=2)``
+
+      This operation will be executed on 4 MPI ranks with 2 OpenMP threads per rank.
+
+    GPU:
+      ``@flow.directives(ngpu=1)``
+
+      The operation requires one GPU for execution.
+
+The :ref:`next section <environments>` provides more details on how to select and possibly define custom environments.
```

Comment:
remove "possibly"

I don't like using "next section" because it makes it hard to remain correct if we change the order of the docs.